WalletWasabi/Services/UpdateManager.cs

```diff
@@ -145,6 +195,49 @@ private async Task<(Version Version, string DownloadUrl, string FileName)> GetLa
 		return (githubVersion, url, fileName);
 	}
 
+	private async Task<bool> ValidateWasabiSignatureAsync(string softwareVersion)
+	{
+		var sha256SumsFilePath = Path.Combine(InstallerDir, "SHA256SUMS.asc");
+		var wasabiSigFilePath = Path.Combine(InstallerDir, "SHA256SUMS.wasabisig");
+
+		using HttpClient httpClient = new();
+		string sha256SumsUrl = $"{DownloadURL}/{softwareVersion}/SHA256SUMS.asc";
+		string wasabiSigUrl = $"{DownloadURL}/{softwareVersion}/SHA256SUMS.wasabisig";
+
+		try
+		{
+			var stream = await httpClient.GetStreamAsync(sha256SumsUrl, CancellationToken).ConfigureAwait(false);
+			await CopyStreamContentToFileAsync(stream, sha256SumsFilePath).ConfigureAwait(false);
+			stream.Close();
+			Sha256SumsFilePath = sha256SumsFilePath;
+
+			stream = await httpClient.GetStreamAsync(wasabiSigUrl, CancellationToken).ConfigureAwait(false);
+			await CopyStreamContentToFileAsync(stream, wasabiSigFilePath).ConfigureAwait(false);
+			stream.Dispose();
+
+			await WasabiSignerHelpers.VerifySha256SumsFileAsync(sha256SumsFilePath).ConfigureAwait(false);
+			return true;
+		}
+		catch (HttpRequestException exc)
+		{
+			string message = "";
+			if (exc.StatusCode is HttpStatusCode.NotFound)
+			{
+				message = "Wasabi signature files was not found under the API.";
+			}
+			else
+			{
+				message = "Something went wrong while geting Wasabi signature files.";
+			}
+			throw new InvalidOperationException(message, exc);
+		}
+		catch (IOException)
+		{
+			CancellationToken.ThrowIfCancellationRequested();
+			throw;
+		}
```

Comment:
I'm not sure if I understand what's happening here.

Reply:
Add a comment.